File: propms/property_management_solution/report/new_repo/new_repo.py
def execute(filters=None):
	if filters.date_from_filter and filters.date_to_filter:
		if filters.date_from_filter > filters.date_to_filter:
			frappe.throw("The 'From Date' ({}) must be before the 'To Date' ({})".format(filters.date_from_filter, filters.date_to_filter))
	if filters.date_from_filter == None:
		filters.date_from_filter = "2000-01-01"
	if filters.date_to_filter == None:
		filters.date_to_filter = frappe.datetime.get_today()
	conditions = " WHERE creation BETWEEN '" + filters.date_from_filter + "' AND '" + filters.date_to_filter + "'"
	if filters.get("name_filter"):
		name = filters.get("name_filter")
		conditions += f" AND user = '{name}'"
	# if len(filters.get("subject_filter")) > 0:
		# subject = ','.join("'{0}'".format(x) for x in filters.get("subject_filter"))
		# conditions += " AND subject IN (" + subject + ")"
	columns = [
		{'fieldname':'name','label':'Name','width':'250'},
		{'fieldname':'full_name','label':'Full Name','width':'250'},
		{'fieldname':'subject','label':'Subject','width':'350'},
		{'fieldname':'status','label':'Status','width':'100'},
		{'fieldname':'creation','label':'Creation','width':'250'}
	]
	data = []
	users = frappe.get_list("User", fields=["name","full_name","'0' AS indent"], filters=[{"name":filters.get("name_filter")}], order_by='full_name ASC')
	for user in users:
		user["has_value"] = True
		data.append(user)
		activities = frappe.db.sql("SELECT subject, status, creation, '1' AS indent FROM `tabActivity Log`" + conditions, as_dict=1)
		for activity in activities:
			activity["has_value"] = False
			data.append(activity)
	return columns, data

File: propms/property_management_solution/report/new_repo/new_repo.js
frappe.query_reports["Script Report Tutorial"] = {
	"treeView": true,
	"name_field": "name",
	"parent_field": "name",
	"initial_depth": 1,			//The level to which the initial rendering will expand to
	//onload: function () {
		//var filter = frappe.query_report.get_filter("name_filter");
		//if (frappe.user.has_role("Administrator") || frappe.user.has_role("System Manager")) {
			//filter.set_input("Goofy");
		//} else {
			//filter.set_input(frappe.user.name);
		//};
		//filter.refresh();
		//frappe.query_report.refresh();
	//},
	"filters": [
		{
			fieldname: "name_filter",
			label: "Name Filter",
			fieldtype: "Link",
			options: "User",
			reqd: 1,
			default: (frappe.user.has_role("Administrator") || frappe.user.has_role("System Manager")) ? "" : frappe.user.name,
			hidden: !(frappe.user.has_role("Administrator") || frappe.user.has_role("System Manager")),
			on_change: function(query_report){
				query_report.set_filter_value('subject_filter', []);
				query_report.refresh();
			},
		},
		{
			fieldname: "subject_filter",
			label: "Subject Filter",
			fieldtype: "MultiSelectList",
			get_data: function(txt) {
				if (frappe.query_report.get_filter_value('name_filter')) {
					var name = frappe.query_report.get_filter_value("name_filter");
					return frappe.db.get_list("Activity Log", {fields: ['subject AS value', 'subject AS label', 'subject AS description'], filters: {"user": name}, distinct: 1, order_by: "subject"});
					//return frappe.db.get_link_options("Activity Log", txt, {"user":name});
				}
				else {
					return [];
				};
			},
			on_change: function(query_report) {
				query_report.refresh();
			},
		},
		{
			fieldname: "date_from_filter",
			label: "Date From Filter",
			fieldtype: "Date",
			default: frappe.datetime.add_months(frappe.datetime.get_today(), -1)
		},
		{
			fieldname: "date_to_filter",
			label: "Date To Filter",
			fieldtype: "Date",
			default: frappe.datetime.get_today()
		}
	],
};
